rule/rule.go

```diff
@@ -1016,6 +1016,15 @@ func (r *Rule) AddArg(value bzl.Expr) {
 	r.updated = true
 }
 
+// UpdateArg replaces an existing arg with a new value.
+func (r *Rule) UpdateArg(index int, value bzl.Expr) {
+	if len(r.args) < index {
+		panic(fmt.Sprintf("Can't update argument at index %d, only have %d args", index, len(r.args)))
```

Comment:
Can we return an error instead of panicking? Some references: https://github.com/uber-go/guide/blob/master/style.md#dont-panic

Reply:
Updated to return an error here, but the only caller is in a non-error-returning context so it still ends up panicking directly